### DISCUSSION TITLE
MPC coefficients calculation method

### DISCUSSION THREAD

**Discussion by hamiddhg** [jorgensd/dolfinx_mpc#108] [General]
Hello,
Is there any reference/document on how the MPC coefficients for periodic boundary conditions are calculated?

**Comment by jorgensd** [comment]
Are you referring to a specific method in DOLFINx_MPC?
For instance: `create_periodic_constraint_geometrical`?

**Reply by hamiddhg** [reply]
Yes, I use: `mpc.create_periodic_constraint_topological`

**Reply by jorgensd** [reply]
`mpc.create_periodic_constraint_topological`
The method starts by finding all degrees of freedom associated with the list of input entities. From those dofs, the dof coordinates are tabulated, and sent into the mapping function, generating a new set of coordinates (those that one should constrain to). These coordinates are then associated with a given (colliding cell), and we compute the contribution of each basis function in this cell.

**Reply by hamiddhg** [reply]
Thank you very much, @jorgensd. It is clear now.
if it's OK with you I'd like to include a rephrased version of your response in my article. Please let me know if there is any reference to cite apart from the present software and FEniCSx citations.

**Reply by jorgensd** [reply]
Hi, that is fine.

To cite DOLFINx_MPC, you should use:
> Dokken, J. S. (2023). DOLFINx-MPC (Version 0.7.0.post0) [Computer software]
or equivalent BIBTEX:

> @software{Dokken_DOLFINx-MPC_2023,author = {Dokken, Jørgen Schartum},month = oct,title = {{DOLFINx-MPC}},version = {0.7.0.post0},year = {2023}}

For DOLFINx you should cite: https://zenodo.org/records/10447666 as mentioned at: https://fenicsproject.org/citing/
